### PR TITLE
Patch 1- Update the external network providers feature page

### PR DESCRIPTION
**Pull request by mburman5**
Update the external network providers feature page with OSP8 as external provider, step by step.

### CODE REVIEW

File: source/develop/release-management/features/external-network-provider.html.md [hunk 1]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
 * get_caps(caps, properties)
 * get_stats(stats, properties)
 
+## Using OSP8 as an external provider
+This feature aims to integrate with any external network provider. We have tested it with OSP8's Neutron using packstack installation, with the following steps and workarounds.
+
+#### Install OSP8 with packstack on Controller, Compute and Networker(the host that will run the network services)
+Please NOTE: This is how we tested, installed and integrated with oVirt(3.6 and 4.0) step by step.
+Some of the configurations, answers in the answerfile and workarounds, maybe relevant only to our spesific or to some other spesisfic use cases.
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: answerfile => answer file
maybe => may be
spesific => specific

File: source/develop/release-management/features/external-network-provider.html.md [hunk 2]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+
+### Controller/Neutron VM
+* Create and run rhel 7.2 VM with 2Gb RAM and 40Gb hardisk
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
+* rhos-release 8
+* yum install openstack-packstack
+* packstack --gen-answer-file=$PACKSTACK_ANSWER_FILE
+* Example to our answerfile can be found --> [answer_file](http://pastebin.test.redhat.com/360085)
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: Example to our answerfile => Example of our answer file here

File: source/develop/release-management/features/external-network-provider.html.md [hunk 3]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* In the neutron VM(controller) we added a second vNIC(in rhev-m) with a VM network and configured in the VM's OS a static IP on this interface(the same subnet as on the compute) for tunneling
+* Enable Network Manager(NM)
+* Run openstack-packstack on the Controller. It will take care of the installation on both the compute and controller
+*  packstack --answer-file=$PACKSTACK_ANSWER_FILE (prepare the compute server before starting the packstack installation)
+
+###Compute Server
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
+* rhos-release 8
+* In our case we configured for tunneling the forth NIC on the server(Compute and networker) with static ip over a non-VM network via Setup networks in rhev-m. It means that the server should already run in rhev-m
+* Delete iptables rule on copmute host --> REJECT all -- anywhere anywhere PHYSDEV match ! --physdev-is-bridged reject-with icmp-host-prohibited --> iptables -D FORWARD 1 (Work Around)
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: copmute => compute

File: source/develop/release-management/features/external-network-provider.html.md [hunk 4]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* yum install vdsm-hook-openstacknet on the compute server
+
+###RHEV-M
+*  Add an external provider in rhev-m -->
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: delete the "-->"

File: source/develop/release-management/features/external-network-provider.html.md [hunk 5]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* Create network/s with subnet/s in neutron VM(controller) and import to rhev-m --> 
+* . keystonerc_admin 
+* neutron net-create 'network_name' 
+* neutron subnet-create 'network_name' 10.0.0.0/24 --name 'subnet_name'  
+* neutron net-list  
+* neutron net-show 'network_id' 
+* Import to rhev-m via 'Networks' main tab or via 'External Providers' sub tab from the left tree
+
+* Run VM/s using this network/s on the compute server 
+*  Currently, a bug discovered when ovirt creating and allocating a new port and binding failing, please see -
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: creating and allocating a new port and binding failing==> is creating and allocating a new port and binding fails

File: source/develop/release-management/features/external-network-provider.html.md [hunk 6]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+[binding_failed](https://bugzilla.redhat.com/show_bug.cgi?id=1318543)
+* neutron port-list -c id -c name -c status -c device_owner -c binding:host_id  (check the status of the ports)
+* ip netns (neutron dhcp status on compute server)
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: netns ==> nets ??

File: source/develop/release-management/features/external-network-provider.html.md [hunk 7]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+## Using OSP8 as an external provider
+This feature aims to integrate with any external network provider. We have tested it with OSP8's Neutron using packstack installation, with the following steps and workarounds.
+
+#### Install OSP8 with packstack on Controller, Compute and Networker(the host that will run the network services)
+Please NOTE: This is how we tested, installed and integrated with oVirt(3.6 and 4.0) step by step.
+Some of the configurations, answers in the answer file and workarounds, may be relevant only to our specific or to some other specific use cases.
+
+### Controller/Neutron VM
+* Create and run rhel 7.2 VM with 2Gb RAM and 40Gb hardisk
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
```
Comment (sandrobonazzola) [inline source/develop/release-management/features/external-network-provider.html.md]: is the above URL publicly accessible? Why not using RDO?

File: source/develop/release-management/features/external-network-provider.html.md [hunk 8]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* rhos-release 8
+* yum install openstack-packstack
+* packstack --gen-answer-file=$PACKSTACK_ANSWER_FILE
+* Example to our answer file can be found --> [answer_file](http://pastebin.test.redhat.com/360085)
```
Comment (sandrobonazzola) [inline source/develop/release-management/features/external-network-provider.html.md]: is above URL publicly accessible? why not include the answerfile here?

File: source/develop/release-management/features/external-network-provider.html.md [hunk 9]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* In the answer file you can choose and enter the IPs for: Controller, Compute and Networker(the Network host, on which to install the network service such as Compute networking (nova network) or Open Stack Networking)
+* In our case, the VM is used as the neutron controller. The Server is used as both the Compute and the Network server
+* Edit the answer file with your relevant configurations. You can compare with the attached answer file for example
+* In the neutron VM(controller) we added a second vNIC(in rhev-m) with a VM network and configured in the VM's OS a static IP on this interface(the same subnet as on the compute) for tunneling
+* Enable Network Manager(NM)
+* Run openstack-packstack on the Controller. It will take care of the installation on both the compute and controller
+*  packstack --answer-file=$PACKSTACK_ANSWER_FILE (prepare the compute server before starting the packstack installation)
+
+###Compute Server
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
```
Comment (sandrobonazzola) [inline source/develop/release-management/features/external-network-provider.html.md]: same here

**Comment by sandrobonazzola** [comment]
@dankenigsberg can you please review as well?

**Comment by thatdocslady** [comment]
@dankenigsberg @mburman5 can you please complete the review and update the PR so we can merge this?

**Comment by sandrobonazzola** [comment]
@dankenigsberg @mburman5 can you please complete the review and update the PR so we can merge this?

File: source/develop/release-management/features/external-network-provider.html.md [hunk 10]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
 * get_caps(caps, properties)
 * get_stats(stats, properties)
 
+## Using OSP8 as an external provider
+This feature aims to integrate with any external network provider. We have tested it with OSP8's Neutron using packstack installation, with the following steps and workarounds.
+
+#### Install OSP8 with packstack on Controller, Compute and Networker(the host that will run the network services)
+Please NOTE: This is how we tested, installed and integrated with oVirt(3.6 and 4.0) step by step.
+Some of the configurations, answers in the answer file and workarounds, may be relevant only to our specific or to some other specific use cases.
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: Delete:Some of the configurations, answers in the answer file and workarounds, may be relevant only to our specific or to some other specific use cases.

Instead we should provide a generic enough description later on.

File: source/develop/release-management/features/external-network-provider.html.md (hunk 8 above)
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: Example to our answer file can be found
=>
A sample anser file:

File: source/develop/release-management/features/external-network-provider.html.md [hunk 11]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+
+### Controller/Neutron VM
+* Create and run rhel 7.2 VM with 2Gb RAM and 40Gb hardisk
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
+* rhos-release 8
+* yum install openstack-packstack
+* packstack --gen-answer-file=$PACKSTACK_ANSWER_FILE
+* Example to our answer file can be found --> [answer_file](http://pastebin.test.redhat.com/360085)
+* In the answer file you can choose and enter the IPs for: Controller, Compute and Networker(the Network host, on which to install the network service such as Compute networking (nova network) or Open Stack Networking)
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: =>
Modify the following entries in the answer file:
- controller IP
- compute IP
- networker IP (the Network host, on which to install the network service such as Compute networking (nova network) or Open Stack Networking)

File: source/develop/release-management/features/external-network-provider.html.md [hunk 12]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* In our case, the VM is used as the neutron controller. The Server is used as both the Compute and the Network server
+* Edit the answer file with your relevant configurations. You can compare with the attached answer file for example
+* In the neutron VM(controller) we added a second vNIC(in rhev-m) with a VM network and configured in the VM's OS a static IP on this interface(the same subnet as on the compute) for tunneling
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: we added -> add
configured in the VM's OS a static IP on this interface = > configured with a static IP

File: source/develop/release-management/features/external-network-provider.html.md [hunk 13]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* Enable Network Manager(NM)
+* Run openstack-packstack on the Controller. It will take care of the installation on both the compute and controller
+*  packstack --answer-file=$PACKSTACK_ANSWER_FILE (prepare the compute server before starting the packstack installation)
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: Add quotes:
packstack --answer-file=$PACKSTACK_ANSWER_FILE
=>
`packstack --answer-file=$PACKSTACK_ANSWER_FILE`

File: source/develop/release-management/features/external-network-provider.html.md [hunk 14]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+This feature aims to integrate with any external network provider. We have tested it with OSP8's Neutron using packstack installation, with the following steps and workarounds.
+
+#### Install OSP8 with packstack on Controller, Compute and Networker(the host that will run the network services)
+Please NOTE: This is how we tested, installed and integrated with oVirt(3.6 and 4.0) step by step.
+Some of the configurations, answers in the answer file and workarounds, may be relevant only to our specific or to some other specific use cases.
+
+### Controller/Neutron VM
+* Create and run rhel 7.2 VM with 2Gb RAM and 40Gb hardisk
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
+* rhos-release 8
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: Add `` around commands

File: source/develop/release-management/features/external-network-provider.html.md [hunk 15]
```diff
@@ -500,6 +500,71 @@ network provider on the host (unless implemented by the user).
...
+* In the neutron VM(controller) we added a second vNIC(in rhev-m) with a VM network and configured in the VM's OS a static IP on this interface(the same subnet as on the compute) for tunneling
+* Enable Network Manager(NM)
+* Run openstack-packstack on the Controller. It will take care of the installation on both the compute and controller
+*  packstack --answer-file=$PACKSTACK_ANSWER_FILE (prepare the compute server before starting the packstack installation)
+
+###Compute Server
+* rpm -i http://rhos-release.virt.bos.redhat.com/repos/rhos-release/rhos-release-latest.noarch.rpm
+* rhos-release 8
+* In our case we configured for tunneling the forth NIC on the server(Compute and networker) with static ip over a non-VM network via Setup networks in rhev-m. It means that the server should already run in rhev-m
+* Delete iptables rule on compute host - REJECT all -- anywhere anywhere PHYSDEV match ! --physdev-is-bridged reject-with icmp-host-prohibited - iptables -D FORWARD 1 (Work Around)
```
Comment (mmirecki) [inline source/develop/release-management/features/external-network-provider.html.md]: Quote '' the rule to delete

**Comment by sandrobonazzola** [comment]
@mmirecki @dankenigsberg @mburman5 can you please complete the review and update the PR so we can merge this?

**Comment by mburman5** [comment]
Guys, there is nothing to review here anymore. packstack is deprecated for a long time and no one is using it anymore.
From the QE side this is not interesting nor relevant  anymore. Marcin, would you agree? 
Any how i have nothing more to add to this page)

**Pull request closed without merging**